src/reference/asciidoc/dsl.adoc

```diff
@@ -115,28 +115,37 @@ The endpoints are automatically wired together by using direct channels.
 
 [[java-dsl-class-cast]]
 .Lambdas And `Message<?>` Arguments
-IMPORTANT: When using lambdas in EIP methods, the "input" argument is generally the message payload.
+[IMPORTANT]
+====
+When using lambdas in EIP methods, the "input" argument is generally the message payload.
 If you wish to access the entire message, use one of the overloaded methods that take a `Class<?>` as the first parameter.
 For example, this won't work:
 
-====
 [source, java]
 ----
 .<Message<?>, Foo>transform(m -> newFooFromMessage(m))
 ----
-====
 
 This will fail at runtime with a `ClassCastException` because the lambda doesn't retain the argument type and the framework will attempt to cast the payload to a `Message<?>`.
 
 Instead, use:
 
-====
 [source, java]
 ----
 .(Message.class, m -> newFooFromMessage(m))
 ----
 ====
 
+[[java-dsl-class-cast]]
+.Bean Definitions override
+[IMPORTANT]
+====
+The Java DSL can register beans for the object defined in-line in the flow definition, as well as can reuse existing, injected beans.
+In case of the same bean name defined for in-line object and existing bean definition, a `BeanDefinitionOverrideException` is thrown indicating that such a configuration is wrong.
+However when you deal with `prototype` beans, there is no way to detect from the integration flow processor an existing bean definition because every time we call a `prototype` bean from the `BeanFactory` we get a new instance.
+This way a provided instance is used in the `IntegrationFlow` as is without any bean registration and any possible check against existing `prototype` bean definition.
```

Comment:
Add "Such a bean may not be fully initialized."

(or should we add `BF.initializeBean()` even if we don't register it?

Reply:
Please, see the latest commit. I'm not fully like it because we really do that against an existing bean and only the problem that we have an `id` and bean is `prototype`.
It doesn't apply for `singleton` because we really do have existing bean and reuse it as is since we have passed that `noBeanPresentForComponent()`.